spring-integration-mongodb/src/main/java/org/springframework/integration/mongodb/inbound/MongoDbMessageSource.java

```diff
@@ -175,6 +177,7 @@ public String getComponentType() {
 	protected void onInit() throws Exception {
 		this.evaluationContext =
 					ExpressionUtils.createStandardEvaluationContext(this.getBeanFactory());
+		this.evaluationContext.setTypeConverter(new StandardTypeConverter(new DefaultConversionService()));
```

Comment:
No, this isn't good. With this way we lose all the customization provided via the common shared config for `EvaluationContext`. 

I'm that won't hurt to evaluate just to `Object` and check it to the supported `String` and `BasicQuery` types. And of course convert String appropriately. 

That's how we will still pursue the backward compatibility and flexibility.

Thank you for the update any way! Will take a look tomorrow closely. Right, that looks pretty much possible to backport the feature to 4.3.x